mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -549,21 +487,76 @@ fn get_p2p_key(ctx: &MmArc, i_am_seed: bool) -> P2PResult<[u8; 32]> {
     Ok(p2p_key)
 }
 
-pub async fn init_p2p(ctx: MmArc) -> P2PResult<()> {
-    let i_am_seed = ctx.is_seed_node();
+fn p2p_precheck(ctx: &MmArc) -> P2PResult<()> {
+    let is_seed_node = ctx.is_seed_node();
+    let is_bootstrap_node = ctx.is_bootstrap_node();
+    let disable_p2p = ctx.disable_p2p();
+    let p2p_in_memory = ctx.p2p_in_memory();
     let netid = ctx.netid();
 
     if DEPRECATED_NETID_LIST.contains(&netid) {
         return MmError::err(P2PInitError::InvalidNetId(NetIdError::Deprecated { netid }));
     }
 
+    let seednodes = seednodes(ctx)?;
+
+    if is_bootstrap_node && !is_seed_node {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Bootstrap node must also be a seed node.".to_owned(),
+        });
+    }
+
+    if is_bootstrap_node && !seednodes.is_empty() {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Bootstrap node cannot have seed nodes to connect.".to_owned(),
+        });
+    }
+
+    if !is_bootstrap_node && seednodes.is_empty() && !disable_p2p {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Non-bootstrap node must have seed nodes configured to connect.".to_owned(),
+        });
+    }
+
+    if disable_p2p && !seednodes.is_empty() {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Cannot disable P2P while seed nodes are configured.".to_owned(),
+        });
+    }
+
+    if disable_p2p && p2p_in_memory {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Cannot disable P2P while using in-memory P2P mode.".to_owned(),
+        });
+    }
+
+    if disable_p2p && is_seed_node {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Seed nodes cannot disable P2P.".to_owned(),
+        });
+    }
+
+    Ok(())
+}
+
+pub async fn init_p2p(ctx: MmArc) -> P2PResult<()> {
+    p2p_precheck(&ctx)?;
+
+    if ctx.disable_p2p() {
+        warn!("P2P is disabled. Features that require a P2P network (like swaps, peer health checks, etc.) will not work.");
+        return Ok(());
+    }
+
+    let is_seed_node = ctx.is_seed_node();
+    let netid = ctx.netid();
+
     let seednodes = seednodes(&ctx)?;
 
     let ctx_on_poll = ctx.clone();
 
-    let p2p_key = get_p2p_key(&ctx, i_am_seed)?;
+    let p2p_key = get_p2p_key(&ctx, is_seed_node)?;
```

Comment:
the value `i_am_seed` was renamed to `is_seed_node`. but we forgot to do it for param in `get_p2p_key` function
```rust
fn get_p2p_key(ctx: &MmArc, i_am_seed: bool) 
```

Reply:
what is this good for? people who don't follow github issues but run seeds will not know about this change...

Reply:
> what is this good for? people who don't follow github issues but run seeds will not know about this change...

What do you mean? Do we have to worry about about users when we rename variables?

Reply:
i mean what is such a rename good for? if people who run seed nodes need to change their config for the seeds to continue to work, then it's not good

Reply:
This is only an internal name change in the source code, it's not the configuration option.

Reply:
@cipig this is an internal code readability improvement. The logic and behavior remain unchanged.
We just renamed the function to better reflect a natural, human-readable question

```rust
pub fn is_seed_node(&self) -> bool { self.conf["i_am_seed"].as_bool().unwrap_or(false) }
```
leads to this
```rust
    let is_seed_node = ctx.is_seed_node();
    let netid = ctx.netid();

    let seednodes = seednodes(&ctx)?;

    let ctx_on_poll = ctx.clone();

    let p2p_key = get_p2p_key(&ctx, is_seed_node)?;
```

Reply:
for me this note is resolved. will leave it open for cipig

Reply:
sorry, i misunderstood the change... thought that the config parameter name is changed